src/ccache.cpp

```diff
@@ -2936,6 +2948,73 @@ cc_process_args(struct args* args,
       continue;
     }
 
+    if (str_eq(argv[i], "--ccache-MJ")) {
+      i++;
+      if (i == argc) {
+        cc_log("Missing argument to --ccache-MJ");
+        result = false;
+        goto out;
+      }
+      output_ccmd = x_strdup(argv[i]);
+      output_ccmd_auto = false;
+      exitfn_add_nullary(write_compile_command_json_if_requested);
```

Comment:
Instead of adding a lot of write_ccmd() calls in various paths I decided to use the exit function.